include/swift/AST/DiagnosticsSema.def

```diff
@@ -760,12 +760,17 @@ ERROR(invalid_redecl,none,"invalid redeclaration of %0", (DeclName))
 ERROR(invalid_redecl_init,none,
       "invalid redeclaration of synthesized %select{|memberwise }1%0",
       (DeclName, bool))
+ERROR(invalid_redecl_implicit,none,
+      "invalid redeclaration of synthesized %select{%0|witness for protocol requirement}1 %2",
```

Comment:
```suggestion
      "invalid redeclaration of synthesized %select{%0|implementation for protocol requirement}1 %2",
```

I think we try to avoid the word 'witness' in user-facing diagnostics.

Reply:
I was trying to think of another word for it! "implementation" seems like a much better choice. Thanks!